source/extensions/filters/network/ext_authz/ext_authz.cc

```diff
@@ -21,6 +23,19 @@ InstanceStats Config::generateStats(const std::string& name, Stats::Scope& scope
                                   POOL_GAUGE_PREFIX(scope, final_prefix))};
 }
 
+void Filter::initializeReadFilterCallbacks(Network::ReadFilterCallbacks& callbacks) {
+  filter_callbacks_ = &callbacks;
+  filter_callbacks_->connection().addConnectionCallbacks(*this);
+
+  // When checking on new connection with a tcp proxy filter we need to prevent the proxy filter
+  // from disabling reads at initialization time
+  if (config_->checkOnNewConnection()) {
+    filter_callbacks_->connection().streamInfo().filterState()->setData(
```

Comment:
This may conflict with an existing key AFAIK - maybe check for presence first and explain what happens if some other filter set it.

Reply:
Filter state is no longer necessary (see comment below)